### PR TITLE
Documentation for immutable tasks

### PR DESCRIPTION
**Pull request by wolfs**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -84,6 +84,42 @@ Text resources like a common checkstyle configuration file can now be fetched di
...
+
+    tasks.create('myTask', CustomTask, 'hello', 42)
+
+In a Groovy build script, you can create the task using `constructorArgs`.
+
+    task myTask(type: CustomTask, constructorArgs: ['hello', 42])
+
+In a Kotlin build script, you can pass constructor arguments using the reified extension function on the `tasks` `TaskContainer`.
+
+    tasks.create<CustomTask>("myTask", "hello", 42)
```
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: Kudos for adding a Kotlin DSL sample 👍

Comment (wolfs) [reply subprojects/docs/src/docs/release/notes.md]: The kudos belongs to @acourtneybrown who originally wrote the documentation. Anyway, the API is different from Kotlin (additional syntactic sugar), so we needed to add the sample.

**Comment by k-mack** [comment]
@wolfs can `null` be injected? Some people ran into issues with the worker API when they tried to inject `null` into a worker's constructor (#2405). Any documentation on the limitations of what can be injected would be good (e.g., "`null` cannot be passed to a task's constructor").

**Comment by wolfs** [comment]
@k-mack Passing null is not possible, see https://github.com/gradle/gradle/pull/5294/files#diff-b91ae463376283a36cfa071f4aa27495R218

**Pull request merged**